pkg/provider/catalogsources.go

```diff
@@ -0,0 +1,72 @@
+package provider
+
+import (
+	"context"
+	"strconv"
+	"strings"
+
+	olmv1Alpha "github.com/operator-framework/api/pkg/operators/v1alpha1"
+	"github.com/redhat-best-practices-for-k8s/certsuite/internal/clientsholder"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type CatalogSource struct {
+	*olmv1Alpha.CatalogSource
+}
+
+func NewCatalogSource(cs *olmv1Alpha.CatalogSource) *CatalogSource {
+	return &CatalogSource{
+		CatalogSource: cs,
+	}
+}
+
+func (cs *CatalogSource) GetBundleCount(env *TestEnvironment) (int, error) {
+	const (
+		grpCurlVersion  = "1.8.5"
+		grpCurlFileName = "grpcurl_" + grpCurlVersion + "_linux_x86_64.tar.gz"
+	)
+
+	o := clientsholder.GetClientsHolder()
+
+	// The index image needs to be queried to get the bundle count.
+	// We accomplish this by exec'ing into the running pod.
+
+	// Get all pods in the cluster wide
+	allPods, err := o.K8sClient.CoreV1().Pods("").List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return -1, err
+	}
+
+	// Look through all of the pods to find the find the pod that is running the index image
+	// If the pod is running an image that is allowlisted, then we can skip it
+
+	// First find the pod that is running the index image
+	for p := range allPods.Items {
+		for c := range allPods.Items[p].Spec.Containers {
+			if allPods.Items[p].Spec.Containers[c].Image != cs.Spec.Image {
+				continue
+			}
+
+			// Found the pod that is running the index image
+			// Now exec into the pod and run the command to get the bundle count
+
+			ctx := clientsholder.NewContext(allPods.Items[p].Namespace,
+				allPods.Items[p].Name, allPods.Items[p].Spec.Containers[c].Name)
+
+			grpCurlURL := "https://github.com/fullstorydev/grpcurl/releases/download/v" + grpCurlVersion + "/" + grpCurlFileName
+
+			createBundlesCommand := "curl -s -L0 " + grpCurlURL + " -o " + grpCurlFileName + "; tar -xf " + grpCurlFileName +
+				"; " + "./grpcurl -plaintext localhost:50051 api.Registry.ListBundles > bundles.txt; cat bundles.txt | grep bundlePath | wc -l"
+
+			// exec into the pod and run the commands
+			cmdValue, errStr, err := o.ExecCommandContainer(ctx, createBundlesCommand)
```

Comment:
Can we output the result of this command execution? I was supposing that this bundleCount was a field within the CatalogSource but now I can see that it's extracted with some commands. So it would be nice to have some logs to debug what's output'ed here